wagtail_live/models.py

```diff
@@ -59,6 +58,168 @@ class LivePageMixin(models.Model):
+            curr_id = self.live_posts[mid].id
+
+            if float(curr_id) == float(live_post_id):
+                return mid
+
+            elif float(curr_id) < float(live_post_id):
+                low = mid + 1
+
+            else:
+                high = mid - 1
+        
+        if self.live_posts[low].id != live_post_id:
+            return -1
+        return low
+
+    def get_live_post_index(self, live_post_id):
+        """Retrieve index of a livepost with its ID.
+        
+        Args:
+            live_post_id (str); Id of the live post to look for.
+
+        Returns:
+            (int) Index of the live post if found else -1
+        """
+
+        return self._get_live_post_index(live_post_id, 0, len(self.live_posts))
+
+    def get_live_post_by_index(self, live_post_index):
+        """Retrieve a live post by its index.
+        
+        Args: 
+            live_post_index (str): Index of the live post to look for.
+
+        Returns:
+            (LivePostBlock) The live post instance
+
+        Raises:
+            (someException) if a live post with the given index doesn't exist.
+        """
+
+        try:
+            live_post = self.live_posts[live_post_index]
+        except KeyError:
+            raise 
+        return live_post
+
+    def get_live_post_by_id(self, live_post_id):
+        """Retrieve a live post by its ID.
+        
+        Args: 
+            live_post_id (str): ID of the live post to look for.
+
+        Returns:
+            (LivePostBlock) The live post instance
+
+        Raises:
+            (someException) if a live post with the given ID doesn't exist.
+        """
+
+        live_post_index = self.get_live_post_index(live_post_id=live_post_id)
+        if live_post_index == -1:
+            raise
+        return self.get_live_post_by_index(live_post_index)
+
+    def add_block_to_live_post(self, block_type, block, live_block):
+        """ Add a new content block to a live post.
+        
+        Args:
+            block_type (str):
+                Type of the block to add
+            block (Block):
+                Block to add to the live post.
+            live_block (LivePostBlock):
+                Live post in which the new block will be added.
+        """
+
+        live_block["content"].append((block_type, block))
+
+    def add_live_post(self, live_post, live_post_id):
+        """ Add a new live post to live page.
+
+        Live posts are added with a custom ID to facilitate
+        keeping their track. 
+        By default, the value of a live post's ID is
+        the timestamp of the corresponding message creation. 
+        A check is also done to ensure that live posts are always
+        sorted by time.
+
+        Args:
+            live_post (LivePostBlock):
+                live post to add
+            live_post_id (str):
+                Id of the new live_post
+        """
+
+        lp_index = len(self.live_posts) if self.live_posts else 0
```

Comment:
We already assume here that self.live_posts is a list, so just using `len(self.live_posts)` will cover every case I think.